### ISSUE TITLE
Feature: allow us to use api tokens 

### ISSUE THREAD

**Issue by ollydixon** [mpociot/chatgpt-vscode#28]
I wrote an Alfred script for chatgpt; so I know it’s possible to use the api key instead of the session token. Much more reliable.

**Comment by timkmecl** [comment]
Hi @ollydixon, what exactly do you mean by api token? The official OpenAI API key or something else?

**Comment by ollydixon** [comment]
In OpenAI we have API keys which how you're supposed to use their APIs.

Using the session key might get your account banned.

<img width="1159" alt="Screenshot 2023-01-03 at 19 54 57" src="https://user-images.githubusercontent.com/6940726/210422896-2cac17cc-3c2d-4bb1-9701-b7d3af8fb063.png">

**Comment by timkmecl** [comment]
You're right, that's the correct/official way of using OpenAI API, and I have a version of this extension that uses this method to query GPT3 [here](https://github.com/timkmecl/codegpt) (but it does not always perform as well as ChatGPT).

However, there is currently no official way to connect to ChatGPT as opposed to GPT3 via their API, so we have to use some unofficial ones until they release it, hopefully soon.

**Comment by ollydixon** [comment]
You can connect to ChatGPT 3

Here's an example using the API key.
```

curl https://api.openai.com/v1/completions \
  -H 'Content-Type: application/json' \
  -H "Authorization: Bearer [api key]" \
  -d '{
  "model": "text-davinci-003",
  "prompt": "${query}",
  "max_tokens": 4000,
  "temperature": 0.9 
}' | /opt/homebrew/bin/jq '.choices[].text' | cut -c6- | sed 's/.$//' > temp.txt 
```

**Comment by timkmecl** [comment]
Your code does not connect to the same model that is available online via chat interface. The code you posted connects to code-davinci, a GPT3.5 model (that is used by my extension I linked above), but not ChatGPT which is a finetuned model based on it.

**Comment by ollydixon** [comment]
> Your code does not connect to the same model that is available online via chat interface. The code you posted connects to code-davinci, a GPT3.5 model (that is used by my extension I linked above), but not ChatGPT which is a finetuned model based on it.

Oh I didn't know

**Comment by timkmecl** [comment]
No problem! After the release of ChatGPT, many people online started calling Gpt3 ChatGpt to join the hype😆 That's why it's easy to mix them up

**Comment by timkmecl** [comment]
@ollydixon  **Try latest update of [my version's repo](https://github.com/timkmecl/chatgpt-vscode) (*the updated version is already on the Marketplace so just update it if you already have it installed*)**

It uses a method described in [this tweet](https://twitter.com/GodlyIgnorance/status/1620270384150093825) (simple Node.js implementation [here](https://github.com/timkmecl/chatgpt-api-wrapper)), already implemented in the unofficial chatgpt api. **Only the official [API key from OpenAI](https://platform.openai.com/account/api-keys) needed.**

This is exactly what you wanted I think
